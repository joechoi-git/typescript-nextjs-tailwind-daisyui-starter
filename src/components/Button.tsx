import React from "react";

interface Props {
    children?: React.ReactNode;
    className?: string;
    [rest: string]: any;
}

export default function Button({ children, className, ...rest }: Props): React.JSX.Element {
    return (
        <button className={`btn btn-primary ${className ? className : ""}`} {...rest}>
            {children}
        </button>
    );
}
