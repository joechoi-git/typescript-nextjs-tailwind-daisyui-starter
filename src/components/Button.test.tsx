import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, jest } from "@jest/globals";
import Button from "./Button";

describe("Button component", () => {
    // eslint-disable-next-line jest/expect-expect
    it("renders without crashing", () => {
        render(<Button />);
    });

    it("renders with children text", () => {
        render(<Button>Click me</Button>);
        expect(screen.getByText("Click me")).toBeInTheDocument();
    });

    it("applies the className correctly", () => {
        render(<Button className="custom-class">Click me</Button>);
        const buttonElement = screen.getByText("Click me");
        expect(buttonElement).toHaveClass("btn btn-primary custom-class");
    });

    it("passes other props correctly", () => {
        const onClick = jest.fn();
        render(<Button onClick={onClick}>Click me</Button>);
        const buttonElement = screen.getByText("Click me");
        fireEvent.click(buttonElement);
        expect(onClick).toHaveBeenCalled();
    });

    it("renders correctly with additional attributes", () => {
        render(<Button data-testid="custom-button">Click me</Button>);
        const buttonElement = screen.getByTestId("custom-button");
        expect(buttonElement).toBeInTheDocument();
    });

    it("matches the snapshot", () => {
        const { asFragment } = render(<Button />);
        expect(asFragment()).toMatchSnapshot();
    });
});
