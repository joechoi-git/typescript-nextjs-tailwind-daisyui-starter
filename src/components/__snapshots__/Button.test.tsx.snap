// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary "
  />
</DocumentFragment>
`;
